qiskit_experiments/curve_analysis/scatter_table.py

```diff
@@ -9,185 +9,497 @@
+
+    * The `data_uid` is an integer key representing the class of the data.
+      When an analysis consists of multiple fit models and performs the multi-objective fit,
+      the created table may contain multiple datasets for each fit model.
+      Usually the index of data matches with the index of the fit model in the analysis.
+      The table also provides `name` column which is a human-friendly text notation of the data_uid.
+      The name and corresponding data_uid must refer to the identical data class,
+      and the name typically matches with the name of the fit model.
+      You can find a particular data subset by either data_uid or name.
+
+    * The `category` is a string key representing a tag of data groups.
+      The measured outcomes input as-is to the curve analysis are tagged with "raw".
+      In a standard :class:`.CurveAnalysis` subclass, the input data is pre-processed for
+      the fitting and the formatted data is also stored in the table with "formatted" tag.
+      After the fit is successfully conducted and the model parameters are identified,
+      data points in the interpolated fit curves are also stored with "fitted" tag
+      for visualization. The management of data group depends on the design of
+      the curve analysis protocol, and the convention of category naming might
+      be different in a particular analysis.
 
-    This class is implemented upon the pandas dataframe.
-    See `pandas dataframe documentation <https://pandas.pydata.org/docs/index.html>`_
-    for the base class API documentation.
+    * The `analysis` is a string key representing a name of
+      the analysis instance that generated the data point.
+      This allows a user to combine multiple tables from the different analyses
+      without collapsing the data points.
+      In the :class:`.CompositeCurveAnalysis`, the instance consists of statistically
+      independent fit models represented in a form of nested component analysis instances.
+      Such component has unique analysis name, and datasets generated from each instance
+      are merged into a single table stored in the outermost composite analysis.
 
-    A single ``ScatterTable`` object can contain different kind of intermediate data
-    generated through the curve fitting, which are classified by the fit model.
-    When an experiment has sub-data for ``sub_exp_1``, the formatted x, y, and y-error
-    array data may be obtained from the original table object as follows:
+    User must be aware of this triplet to extract data points that belong to a
+    particular data subset. For example,
 
-    .. code-block::python
+    .. code-block:: python
 
-        abc_data = table[
-            (table.name == "sub_exp_1") & (table.category == "formatted")
-        ]
-        x, y, e = abc_data.xval.to_numpy(), abc_data.yval.to_numpy(), abc_data.yerr.to_numpy()
+        mini_table = table.filter(kind="model1", category="raw", analysis="Analysis_A")
```

Comment:
I find using `kind` as a filter name a bit confusing since it means the same as category. How about `model` or `class` instead?

Reply:
Replaced with `data_uid`. Usually it matches with the model but some case not. For example, in the StarkRamseyXYAmpScanAnalysis in the raw category this field represents either X or Y measurement data but after formatted the field represents frequency data in the fit model. class is reserved word.